plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -1801,19 +1803,28 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
 
     -- Conversions
     toBuiltinMeaning _semvar IntegerToByteString =
-      let integerToByteStringDenotation :: Bool -> Integer -> Integer -> BuiltinResult BS.ByteString
-          integerToByteStringDenotation = integerToByteStringWrapper
+      let integerToByteStringDenotation :: Bool -> LiteralByteSize -> Integer -> BuiltinResult BS.ByteString
+          {- The second argument is wrapped in a LiteralByteSize to allow us to interpret it as a size during
+             costing.  It appears as an integer in UPLC: see Note [Integral types as Integer]. -}
+          {- We disallow inputs of more than 8000 bytes because the execution time becomes difficult to predict
+             accurately beyond this point.  This restriction will be removed in a later variant. -}
+          integerToByteStringDenotation b (LiteralByteSize w) n =
+              if w > 8000 || (w == 0 && (sumCostStream . flattenCostRose . memoryUsage $ n) > 1000)
```

Comment:
This is a bit annoying, but it has to go here because a later implementation may be able to dispense with this check, and then we'll need two variants and two denotations.

Reply:
The `sumCostStream . flattenCostRose . memoryUsage` pattern appears in several places in the codebase ...